controllers/auth/index.js

```diff
@@ -1,35 +1,35 @@
 import express from "express";
 import { ApiError } from "../../utils/ApiError.js";
 import { findUserByEmail } from "../../services/user/index.js";
+import generateToken from "../../services/auth/index.js";
+import bcrpyt from "bcrypt";
 
 const router = express.Router();
 
 // TODO: develop the APIs and add the swagger documentation here using JSDoc syntax
 
-router.post("/login", (req, res) => {
+router.post("/login", async (req, res) => {
   try {
     const { email, password } = req.body;
 
-    //user validation ( email, password)
-
     if (!email) throw new ApiError(400, "Email is required");
 
     const user = findUserByEmail(email);
     if (!user) throw new ApiError(404, "User doesn't exist");
 
-    const isPasswordValid = user.isPasswordCorrect(password);
+    const isPasswordValid = await bcrpyt.compare(password, user?.password);
     if (!isPasswordValid) throw new ApiError(401, "Invalid user credentials");
 
     //generate access Token
-    const accessToken = user.generateAccessToken();
+    const accessToken = generateToken(user?._id);
```

Comment:
I see that you've created a new `generateToken` function, but I went throw PR #6 seems like the developer has already made a service for this, can you take a pull and utilize that, this will be an opportunity for you to understand collaboration.

Reply:
PR #6 has not merged yet, is it ok to merge it into my branch?